Support/Scripts/generate-documentation.sh

```diff
@@ -0,0 +1,65 @@
+#!/usr/bin/env bash
+
+# This is used to automatically generate documentation for the project and is
+# inspired by https://gist.github.com/domenic/ec8b0fc8ab45f39403dd
+
+set -eu
+
+SOURCE_BRANCH="master"
+TARGET_BRANCH="gh-pages"
+DEPLOY_KEY="Support/Testing/Travis/deploy_key.enc"
+TARGET_DIRECTORY="html"
+
+do_nothing() {
+  echo "exiting: $1"
+  exit 0
+}
+
+REPO="$(git config remote.origin.url)"
+SSH_REPO="${REPO/https:\/\/github.com\//git@github.com:}"
+SHA="$(git rev-parse --verify HEAD)"
+CURRENT_BRANCH="$(git rev-parse --abbrev-ref HEAD)"
+
+if [ ! -d "$TARGET_DIRECTORY" ]; then
+  git clone --depth 1 --branch "$TARGET_BRANCH" "$REPO" "$TARGET_DIRECTORY"
```

Comment:
Good point.

On Thursday, June 16, 2016, Francis Ricci <notifications@github.com> wrote:

> In Support/Scripts/generate-documentation.sh
> <https://github.com/facebook/ds2/pull/374#discussion_r67451692>:
>
> > +TARGET_BRANCH="gh-pages"
> > +DEPLOY_KEY="Support/Testing/Travis/deploy_key.enc"
> > +TARGET_DIRECTORY="html"
> > +
> > +do_nothing() {
> > +  echo "exiting: $1"
> > +  exit 0
> > +}
> > +
> > +REPO="$(git config remote.origin.url)"
> > +SSH_REPO="${REPO/https:\/\/github.com\//git@github.com:}"
> > +SHA="$(git rev-parse --verify HEAD)"
> > +CURRENT_BRANCH="$(git rev-parse --abbrev-ref HEAD)"
> > +
> > +if [ ! -d "$TARGET_DIRECTORY" ]; then
> > +  git clone --depth 1 --branch "$TARGET_BRANCH" "$REPO" "$TARGET_DIRECTORY"
>
> why not git_clone?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/facebook/ds2/pull/374/files/516880511fbd6980998450bee02966d8eaf6e926..c62d425c01d829cd323074127b1b18109a4c8a46#r67451692>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AADHbG2L3pHQFjnRniD3Nuyr0AVHjyGPks5qMf2NgaJpZM4I22ku>
> .
>


-- 
Stephane Sezer